src/main/java/org/jabref/logic/citation/repository/LRUBibEntryRelationsCache.java

```diff
@@ -0,0 +1,57 @@
+package org.jabref.logic.citation.repository;
+
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.identifier.DOI;
+
+import org.eclipse.jgit.util.LRUMap;
+
+public class LRUBibEntryRelationsCache {
+    private static final Integer MAX_CACHED_ENTRIES = 100;
+    private static final Map<DOI, Set<BibEntry>> CITATIONS_MAP = new LRUMap<>(MAX_CACHED_ENTRIES, MAX_CACHED_ENTRIES);
+    private static final Map<DOI, Set<BibEntry>> REFERENCES_MAP = new LRUMap<>(MAX_CACHED_ENTRIES, MAX_CACHED_ENTRIES);
+
+    public List<BibEntry> getCitations(BibEntry entry) {
+        return entry
+            .getDOI()
+            .stream()
+            .flatMap(doi -> CITATIONS_MAP.getOrDefault(doi, Set.of()).stream())
+            .toList();
+    }
+
+    public List<BibEntry> getReferences(BibEntry entry) {
+        return entry
+            .getDOI()
+            .stream()
+            .flatMap(doi -> REFERENCES_MAP.getOrDefault(doi, Set.of()).stream())
+            .toList();
+    }
+
+    public void cacheOrMergeCitations(BibEntry entry, List<BibEntry> citations) {
```

Comment:
Method used to rewrite data, now it does merge inputs according the method name.